plugins/modules/zos_copy.py

```diff
@@ -2447,6 +2478,34 @@ def run_module(module, arg_def):
         )
 
     # ********************************************************************
+    # To validate the source and dest are not lock in a batch process by
+    # the machine and not generate a false positive check the disposition
+    # for try to write in dest and if both src and dest are in lock.
+    # ********************************************************************
+    if src_ds_type != "USS" and dest_ds_type != "USS":
+        is_source_lock = source_locked(src_name)
+        is_dest_lock = source_locked(dest_name)
+        if not is_source_lock and not is_dest_lock:
+            module.fail_json(
+                msg="DATASETS in lock, unable to access'{0}' without force and unable to write in'{1}'".format(
+                    src_name, dest_name
+                )
+            )
+        elif not is_dest_lock:
+            module.fail_json(
+                msg="DATASET in lock, unable to wrote in '{0}'".format(
+                    dest_name
+                )
+            )
+    elif dest_ds_type != "USS":
```

Comment:
The case of a dataset in DSP=SHR works for read not for writing and if both are in DSP=SHR came a weird error that lock both datasets and you couldn't access to both datasets for some reason.

Reply:
that's why I asked.

Reply:
When I was looking over this code, I questioned the reason to check if the src is locked and then went back to my notes which I will share here, I believe (i would have to test) that the issue you saw with the error on the source i fit is locked is accurate. Likely the default since we don't support SHR is that data sets are opened as OLD or MOD, thus an exclusive lock, thus with either OLD/MOD no one can access the data set and hence why you need to also check the source, this is good, thanks @AndreMarcel99 , fiy @richp405 

```
OLD - file must exist and you have exclusive control/hold of the dataset, that is, nobody else can read the file while you are using it. You will read from beginning of dataset and if you write, then it will overwrite on existing data. i.e old data is lost.
SHR - file must exist and all others may READ (multiple user can share the data, such then its a read-only file) it at the same time you are reading it (they must have SHR as well)
MOD - file must exist and you have exclusive control/hold of the dataset, will be adding to the end of the data set, without loosing your old data (i.e append the data to the existing file).
NEW - you are creating the data set for the 1st time.
```

Reply:
I had resolved because I got an answer to my question... I thought we could read if dsp=shr, so the source query he had was legit.  The source you put up does the same thing, and looks cleaner to me.

Reply:
I did some testing, i will put my steps here as i had share in slack as well:

My test:

SSH , setup:
```
dtouch ibmuser.pdse.a
dtouch ibmuser.pdse.b
decho 'hello' 'ibmuser.pdse.a(one)'
decho 'hello' 'ibmuser.pdse.b(one)'
```

ISPF:
1. ISPF 3.4 > ibmuser.pdse.a
2. e next to ibmuser.pdse.a
3. FN F2 (split screen) 
4. ISPF 3.4 > ibmuser.pdse.b
5. e next to ibmuser.pdse.b

**Copy test 1- Both src & dest locked:**
- `cp "//'ibmuser.pdse.a(one)'" "//'ibmuser.pdse.b(onecopy)'"`
- `cp: FSUM6259 target file “//‘ibmuser.pdse.b(onecopy)’“: EDC5061I An error occurred when attempting to define a file to the system.`

**Copy test 2 - src locked, dest not locked:**
- `cp "//'ibmuser.pdse.a(one)'" "//'ibmuser.pdse.b(onecopy2)'"`
- `Success`

**Copy test 3 - src unlocked, dest locked:**
- `cp "//'ibmuser.pdse.a(one)'" "//'ibmuser.pdse.b(onecopy3)'"`
- `cp: FSUM6259 target file “//‘ibmuser.pdse.b(onecopy3)’“: EDC5061I An error occurred when attempting to define a file to the system.`

**Copy test 4 - src unlocked, dest unlocked:**
- `cp "//'ibmuser.pdse.a(one)'" "//'ibmuser.pdse.b(onecopy4)'"`
- `Success`

What this indicates to me is **only** the `dest`  has to be monitored for a lock, not the `src`, not the `src && dest`, just the `dest`:
- There is no issue with a ISPF locking up if both data sets are in disp=shr (locked) and a copy (cp ) is in progress
- There is no need to have the this double check if src_ds_type != "USS" and dest_ds_type != "USS":
- There is no need to monitor src (no one is doing it, i am just stating that its not needed. 

The reason **Copy test 1** failed is because the `dest` is locked, it had nothing to do with both `src && dest` being locked, its the same reason an error in **Copy test 3 failed**, because both these tests locked the `dest`.

The Pull request will need updating. 

Result of ISPF browse to prove the results match the tests:
![image](https://github.com/ansible-collections/ibm_zos_core/assets/25803172/4bd8a1cf-9b57-462e-b408-150786ef7b84)
![image](https://github.com/ansible-collections/ibm_zos_core/assets/25803172/fd205e26-3680-4c31-a6ab-776b74fdc45c)